### PR TITLE
feat(changelog): do not skip breaking changes if configured

### LINKED ISSUE #106: Allow special handling for "skipped" breaking changes
On many of my projects, I choose to skip some scopes in the changelog, for example `chore` and `refactor`. This keeps the noise down and allows people to see features and fixes without the distraction of all the moving cheese.

However sometimes one of those commits will be marked as a breaking change (`chore!: ....`). These commits should not be skipped in the changelog. I found I catch these with a commit parser added early in the list. Unfortunately the other rules setting group titles then don't get applied.

```toml
commit_parsers = [
    { message = ".*!:", skip = false },
    { message = "^chore", group = "Miscellaneous Tasks", skip = true },
]
```

I can work it out by having duplicate rules for each group with breaking changes not being skipped and others being skipped, but it means a lot more rules that I would have expected.

I think there should probably be some top level support for handling breaking changes such that they can always override skip values.

### PR DESCRIPTION
**Pull request by sbmueller**
Hi there! Trying to contribute to this project first-time.

## Description
Trying to resolve #106 by option to protect breaking changes, even when matched by a skipping commit parser. The option `protect_breaking_commits` is introduced in the config.
Note that during testing, one caveat was discovered: If a commit is matched by one parser, it can't be matched by any other. That means if a breaking commit was prevented from being skipped, it won't become part of any other group. Maybe it is worth to think about allowing multiple parsers for commits in an hierarchical order?

## Motivation and Context

Breaking changes are highly relevant for stakeholders, therefore should not be skipped in a changelog, even when matched by commit parsers.

Fixes #106 

## How Has This Been Tested?

Extended the test cases by breaking change commits. Added a usual breaking change commit and one that matches a skipped commit parser regex. The latter is not skipped even it matches a commit parser that has `skip` set to true, if `protect_breaking_commits` is true.

nightly-aarch64-apple-darwin / rustc 1.66.0-nightly (8ce3204af 2022-09-30)

Backward compatibility is provided.

## Screenshots / Output (if appropriate):
See tests in `src/changelog.rs`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (no code change)
- [ ] Refactor (refactoring production code)
- [ ] Other 

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by orhun — CHANGES_REQUESTED** [review]
LGTM! Just a few edits needed, see my comments 🐻

**Comment by orhun** [comment]
I added test fixtures for this PR in [4e61f30](https://github.com/orhun/git-cliff/pull/114/commits/4e61f30412e4dd3f4780e808831e818f2b2020b3)

File: git-cliff-core/src/config.rs [hunk 1]
```diff
@@ -47,25 +47,28 @@ pub struct GitConfig {
...
 
 	/// Git commit preprocessors.
-	pub commit_preprocessors: Option<Vec<CommitPreprocessor>>,
+	pub commit_preprocessors:     Option<Vec<CommitPreprocessor>>,
 	/// Git commit parsers.
-	pub commit_parsers:       Option<Vec<CommitParser>>,
+	pub commit_parsers:           Option<Vec<CommitParser>>,
+	/// Wether to protect all breaking changes from being skipped by a commit
+	/// parser.
+	pub protect_breaking_commits: Option<bool>,
```
[the remark this answers is not in the document]
Comment (sbmueller) [reply git-cliff-core/src/config.rs:55]: Went with `false` here to maintain backward compatibility

File: git-cliff-core/src/commit.rs [hunk 2]
```diff
@@ -233,13 +236,29 @@ impl Commit<'_> {
 		Ok(self)
 	}
 
+	/// States if the commit is skipped in the provided `CommitParser`.
+	///
+	/// Returns `true` if the parser's `skip` field is `true`. If
+	/// `protect_breaking_commits` is enabled in the config and the commit is
+	/// breaking, or parser's `skip` field is None or `false` returns `false`.
```
[the remark this answers is not in the document]
Comment (sbmueller) [reply git-cliff-core/src/commit.rs:243]: Rephrased it a bit, hope it is better now

File: README.md [hunk 3]
```diff
@@ -593,6 +594,13 @@ Examples:
   - If the commit starts with "doc", group the commit as "Documentation" and set the default scope to "other". (e.g. `docs: xyz` will be processed as `docs(other): xyz`)
 - `{ message = "(www)", scope = "Application"}`
   - If the commit contains "(www)", override the scope with "Application". Scoping order is: scope specification, conventional commit's scope and default scope.
+
```
[the remark this answers is not in the document]
Comment (sbmueller) [reply README.md:597]: Oops, wasn't intentional

**Comment by sbmueller** [comment]
Thanks @orhun for the review! I think I addressed everything now. Feel free to add more comments if unsure about something.

**Review by orhun — APPROVED** [review]
Great stuff, thanks for your contribution! 🐻

**Pull request merged**